pkg/trafficpolicy/trafficpolicy.go

```diff
@@ -56,3 +56,81 @@ func (out *OutboundTrafficPolicy) AddRoute(httpRouteMatch HTTPRouteMatch, weight
 	}
 	return nil
 }
+
+// MergeInboundPolicies merges latest InboundTrafficPolicies into original InboundTrafficPolicies
+func MergeInboundPolicies(original []*InboundTrafficPolicy, latest ...*InboundTrafficPolicy) []*InboundTrafficPolicy {
+	for _, l := range latest {
+		foundHostnames := false
+		for _, or := range original {
+			if reflect.DeepEqual(or.Hostnames, l.Hostnames) {
+				foundHostnames = true
+				or.Rules = mergeRules(or.Rules, l.Rules)
+			}
+		}
+		if !foundHostnames {
+			original = append(original, l)
```

Comment:
I suspect we may end up with duplicate `InboundTrafficPolicy`? Would a map/set make sense?

Reply:
Yeah let me look into that. I've been a bit worries about passing sets around bc you can't specify the type of objects in the set when you're passing if to and from methods.